airbyte-oauth/src/main/java/io/airbyte/oauth/BaseOAuthFlow.java

```diff
@@ -35,7 +39,9 @@ protected JsonNode getSourceOAuthParamConfig(final UUID workspaceId, final UUID
       final Optional<SourceOAuthParameter> param = MoreOAuthParameters.getSourceOAuthParameter(
           configRepository.listSourceOAuthParam().stream(), workspaceId, sourceDefinitionId);
       if (param.isPresent()) {
-        return param.get().getConfiguration();
+        // TODO: if we write a flyway migration to flatten persisted configs in db, we don't need to flatten
+        // here see https://github.com/airbytehq/airbyte/issues/7624
+        return Jsons.flattenConfig(param.get().getConfiguration());
```

Comment:
Because https://github.com/airbytehq/airbyte/issues/5989 is not implemented yet (and https://github.com/airbytehq/airbyte/issues/7624), we need to keep instance wide params in their original form (nested) or the `OAuthConfigSupplier` won't be able to merge params properly into connector configs at runtime.

So we flatten them at runtime for the moment before using them in the OAuth flows.